### PR TITLE
CXP-296: Store exception details inside Elasticsearch

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/ErrorManagement/ExtractErrorsFromHttpException.php [hunk 1]
```diff
@@ -26,9 +26,13 @@ class ExtractErrorsFromHttpException implements ExtractErrorsFromHttpExceptionIn
     /** @var Serializer */
     private $serializer;
 
-    public function __construct(Serializer $serializer)
+    /** @var Serializer */
+    private $legacySerializer;
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/ErrorManagement/ExtractErrorsFromHttpException.php:30]: It's not legacy but `pimSerializer`

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/AkeneoConnectivityConnectionBundle.php [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 namespace Akeneo\Connectivity\Connection\Infrastructure\Symfony;
 
+use Akeneo\Pim\Enrichment\Bundle\DependencyInjection\Compiler\RegisterSerializerPass;
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/AkeneoConnectivityConnectionBundle.php:7]: I didn't see that the RegisterSerializerPass was in enrichment, maybe we can have our own simple Compiler Pass to add the normalizers/encoders to the serializer service. It should be fast to do.

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/DependencyInjection/AkeneoConnectivityConnectionExtension.php [hunk 3]
```diff
@@ -29,5 +29,6 @@ public function load(array $configs, ContainerBuilder $container): void
         $loader->load('repositories.yml');
         $loader->load('controllers.yml');
         $loader->load('event_subscribers.yml');
+        $loader->load('normalizers.yml');
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/DependencyInjection/AkeneoConnectivityConnectionExtension.php:32]:
```suggestion
        $loader->load('serializers.yml');
```

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/config/normalizers.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        class: 'Symfony\Component\Serializer\Serializer'
+
+    akeneo_connectivity.connection.normalizer.exception.violation:
+        class: 'Akeneo\Connectivity\Connection\Infrastructure\Normalizer\ViolationNormalizer'
+        arguments:
+            - '@pim_catalog.repository.cached_attribute'
+        tags:
+            - { name:  akeneo_connectivity.connection.serializer.normalizer, priority: 100 }
+
+    akeneo_connectivity.connection.encoder.exception.violation:
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/config/normalizers.yml:17]:
```suggestion
    akeneo_connectivity.connection.encoder.json:
```

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/ErrorManagement/ExtractErrorsFromHttpException.php (hunk 1 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/ErrorManagement/ExtractErrorsFromHttpException.php:30]: `externalApiSerializer` if we want to be explicit or `fallbackSerializer` for something more logical

**Comment by pierrejolly** [comment]
There is only one class imported from enrichment `RegisterSerializerPass` and the rule is added to `.php_cd` so everything work fine

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/AkeneoConnectivityConnectionBundle.php (hunk 2 above)
Comment (pierrejolly) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/AkeneoConnectivityConnectionBundle.php:7]: As you want, but it's will be an exact copy/paste of the existing one

**Pull request merged**